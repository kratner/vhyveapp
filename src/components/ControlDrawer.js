import React, { Component } from "react";
import { MdDragHandle } from "react-icons/md";
import CameraToggle from "./CameraToggle";
import MediaInputSelector from "./MediaInputSelector";
import styled from "styled-components";

const ControlDrawerContainer = styled.div`
  position: absolute;
  bottom: 0;
  width: 100%;
  z-index: 10;
`;

const DragHandle = styled(MdDragHandle)`
  color: #fff;
`;

const DrawerHandle = styled.div`
  background-color: #ffffff10;
  cursor: pointer;
  text-align: center;
  width: 100%;
  transition: 0.15s;
  &:hover {
    background-color: #ffffff50;
  }
`;

const ControlContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-around;
`;

class ControlDrawer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: true
    };
    this.handleToggle = this.handleToggle.bind(this);
    this.handleCameraSwitch = this.handleCameraSwitch.bind(this);
  }
  handleToggle() {
    const controlDrawer = document.getElementById("controlcontainer");
    this.setState({
      open: !this.state.open
    });
    if (this.state.open) {
      controlDrawer.classList.add("open");
    } else {
      controlDrawer.classList.remove("open");
    }
  }
  handleCameraToggle() {
    this.props.handleCameraToggle();
  }
  handleCameraSwitch() {
    this.props.handleCameraSwitch();
  }
  handleAudioInputItemSelect(thisComponent, DeviceId) {
    this.props.handleAudioInputItemSelect(DeviceId);
  }
  handleVideoInputItemSelect(thisComponent, DeviceId) {
    this.props.handleVideoInputItemSelect(DeviceId);
  }
  render() {
    return (
      <ControlDrawerContainer id="controldrawer" className="controldrawer">
        <DrawerHandle onClick={this.handleToggle} title="Toggle Controls">
          <DragHandle />
        </DrawerHandle>
        <ControlContainer id="controlcontainer">
          <CameraToggle
            size="1.5em"
            active={this.props.cameraActive}
            onClick={() => {
              this.handleCameraToggle();
            }}
          />
          <MediaInputSelector
            onVideoItemClick={DeviceID => {
              this.handleVideoInputItemSelect(this, DeviceID);
            }}
            onAudioItemClick={DeviceID => {
              this.handleAudioInputItemSelect(this, DeviceID);
            }}
          />
        </ControlContainer>
      </ControlDrawerContainer>
    );
  }
}

export default ControlDrawer;
