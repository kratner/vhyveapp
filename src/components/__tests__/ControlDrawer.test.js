import React from "react";
import { shallow } from "enzyme";
import ControlDrawer from "../ControlDrawer";

describe("ControlDrawer", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ControlDrawer />);
    expect(wrapper).toMatchSnapshot();
  });
});
