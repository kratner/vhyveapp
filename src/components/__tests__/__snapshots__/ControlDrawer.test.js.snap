// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlDrawer renders correctly 1`] = `ShallowWrapper {}`;
